internal/workflow.go

```diff
@@ -2017,17 +2018,28 @@ func convertFromPBRetryPolicy(retryPolicy *commonpb.RetryPolicy) *RetryPolicy {
 		NonRetryableErrorTypes: retryPolicy.NonRetryableErrorTypes,
 	}
 
-	// Avoid nil pointer dereferences
-	if v := retryPolicy.MaximumInterval; v != nil {
-		p.MaximumInterval = *v
-	}
-	if v := retryPolicy.InitialInterval; v != nil {
-		p.InitialInterval = *v
-	}
+	p.MaximumInterval = duration.Value(retryPolicy.MaximumInterval)
+	p.InitialInterval = duration.Value(retryPolicy.InitialInterval)
 
 	return &p
 }
 
+func copyPBRetryPolicy(retryPolicy *commonpb.RetryPolicy) *commonpb.RetryPolicy {
```

Comment:
I don't think I'd discovered proto.Clone when I first wrote this, heh.

I'll eventually add shallow `.Copy()` helpers we can use but that won't be for a bit; I need the server to compile first

Reply:
This helper has been removed

Reply:
> I'll eventually add shallow .Copy() helpers we can use but that won't be for a bit; I need the server to compile first

Hopefully not as user-facing API

Reply:
The server makes heavy use of them so they may need to be, and go won't let us extend a type outside of its package so we can't generate them only in the server.

Reply:
Unfortunately with the transition away from gogo we lose the ability to make a shallow copy. Since the server _also_ uses the go API generating that functionality here is easier; otherwise the sanest approach would (imo) be for the server to no longer depend on the API but codegen the API functionality itself so that it could _also_ generate these helper methods.

I don't want the server team to need to maintain these manually